tests/backends/postgresql/tests.py

```diff
@@ -223,6 +223,90 @@ def test_connect_non_autocommit(self):
         finally:
             new_connection.close()
 
+    @unittest.skipUnless(is_psycopg3, "psycopg3 specific test")
+    def test_connect_pool(self):
+        from psycopg_pool import PoolTimeout
+
+        new_connection = connection.copy()
+        self.assertIsNone(new_connection.pool)
+
+        new_connection.settings_dict["OPTIONS"]["pool"] = {
+            "min_size": 2,
+            "timeout": 0.1,
+        }
+        try:
+            self.assertIsNotNone(new_connection.pool)
+            conn_params = new_connection.get_connection_params()
+            connection_1 = new_connection.get_new_connection(conn_params)
+            new_connection.get_new_connection(conn_params)
+            with self.assertRaises(PoolTimeout):
+                # The pool has a maximum of 2 connections.
+                new_connection.get_new_connection(conn_params)
+
+            new_connection.pool.putconn(connection_1)
+            connection_3 = new_connection.get_new_connection(conn_params)
+            # Confirm reused the first connection when one is available.
```

Comment:
To add more explanation, comments should state the expected behavior and omit prefixes like "Confirm" since all tests confirm things. This guideline is from [Python coding style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style).


```suggestion
            # Confirm reused the first connection when one is available.
```